carbonplan_tonyear/core.py

```diff
@@ -0,0 +1,141 @@
+import json
+import math
+
+import numpy as np
+
+
+def get_irf_curve(curve_name):
+    if curve_name == 'joos_2013':
+        # parameters from Joos et al., 2013 (Table 5)
+        # https://doi.org/10.5194/acp-13-2793-2013
+        a = [0.2173, 0.2240, 0.2824, 0.2763]
+        tau = [0, 394.4, 36.54, 4.304]
+        t_horizon = np.arange(1001)
+    elif curve_name == 'ipcc_2007':
+        # parameters from IPCC AR4 2007 (Chapter 2, page 213)
+        # https://www.ipcc.ch/site/assets/uploads/2018/02/ar4-wg1-chapter2-1.pdf
+        a = [0.217, 0.259, 0.338, 0.186]
+        tau = [0, 172.9, 18.51, 1.186]
+        t_horizon = np.arange(1001)
+    elif curve_name == 'ipcc_2000':
+        # parameters from IPCC LULUCF Special Report 2000 (Chapter 2.3.6.3, Footnote 4)
+        # https://archive.ipcc.ch/ipccreports/sres/land_use/index.php?idp=74
+        a = [0.175602, 0.137467, 0.18576, 0.242302, 0.258868]
+        tau = [0, 421.093, 70.5965, 21.42165, 3.41537]
+        t_horizon = np.arange(1001)
+    else:
+        raise ValueError('No IRF parameters by the name \'' + curve_name + '\'.')
+
+    irf_curve = [a[0]] * len(t_horizon)
+    for t in t_horizon:
+        for i in np.arange(1, len(a)):
+            irf_curve[t] = irf_curve[t] + (a[i] * np.exp(-t / tau[i]))
+    return irf_curve
+
+
+def get_discounted_curve(discount_rate, curve):
+    return [x / math.pow(1 + discount_rate, i) for i, x in enumerate(curve)]
+
+
+def print_benefit_report(method_output):
+    discount = str(round(method_output['parameters']['discount_rate'] * 100, 1))
+    delay = str(method_output['parameters']['delay'])
+    baseline_atm_cost = str(round(method_output['baseline_atm_cost'], 2))
+    benefit = str(round(method_output['benefit'], 2))
+    num_needed = str(round(method_output['num_for_equivalence'], 1))
+
+    print()
+    print('Discount rate: ' + discount + '%')
+    print('Delay: ' + delay + ' year(s)')
+    print('Baseline atmospheric cost: ' + baseline_atm_cost + ' ton-years')
+    print('Benefit from 1tCO2 with delay: ' + benefit + ' ton-years')
+    print('Number needed: ' + num_needed)
+    print()
+
+
+def calculate_tonyears(method, baseline, time_horizon, delay, discount_rate):
```

Comment:
the comment makes this clear, but just going to ask one more question. 

this function requires that you provide `baseline`, which is the output of `get_irf_curve`. Is it at all useful to sync the terminology here? E.g., `get_baseline_curve`? 

if this is a good idea to change the name, we should update instances of `irf` to `baseline`

Reply:
I think this a great idea! Changing.